src/psyclone/dynamo0p3.py

```diff
@@ -8277,6 +8277,13 @@ class creates the PSyIR schedule on first invocation which is
 
         # Get the PSyIR Kernel Schedule(s)
         routines = Fparser2Reader().get_routine_schedules(self.name, self.ast)
+        for routine in routines:
+            # If one of the symbols is not declared in a routine then
+            # this is only picked up when writing out the routine
+            # (raising a VisitorError), so we check here so that
+            # invalid code is not inlined. We use debug_string() to
+            # minimise the overhead.
+            routine.debug_string()
```

Comment:
"F2PY_EXPR_TUPLE_1" is an internal fparser2 or fparser1 token that should never appear as a symbol in the PSyIR, so is most likely an internal fparser2 error that needs chasing down. I think that in theory we should never get symbols like this (i.e. with no type info at all - unless using variables without declarations causes this? - I'll check that with an example) so we are really picking up errors - unless you know of other valid cases for valid Fortran - but we have one here and there may be others in the future so it is probably good to continue to pick up here (or even earlier if possible). The problem is that such a symbol is happily inlined into the parent symbol table even though it will never become valid symbol as we now only check for symbols with no type information when writing out the code as we can have cases where type info is hidden in things like unknownfortrantype which only becomes visible in the back-end.

So I don't think _prepare_code_to_inline will help as we don't yet know what types things will be, or conversely if there are things there that will never be types and cause an exception - in the back-end now because we are only checking there.

Given this, I think I'll stick with the hacky debug_string() which I could be change to use FortanWriter if you are happy to take the hit, or we could just call the symbol resolving part of FortranWriter if that is possible and you like to explore that.

I've added #2271 if (as usual) you are correct about this where suggested but have also add a comment along the lines of the above next to in the unlikely event that I am right.

Reply:
If you bring the branch to master we have updated LFRic to trunk, which I think you said it didn't have this issue. If this is the case, maybe you can remove this hotfix and we can continue this discussion in a new issue, as I am not convinced with the solution but I don't want to delay the rest of the PR.

Reply:
I don't think it is to do with the version of LFRic unfortunately.

Reply:
Just checked with the latest version of LFRic and without the "hotfix" we still get ...

```
psyclone.psyir.backend.visitor.VisitorError: Visitor Error: The following symbols are not explicitly declared or imported from a module and there are no wildcard imports which could be bringing them into scope: 'F2PY_EXPR_TUPLE_1'
```